llvm/lib/CodeGen/SelectionDAG/LegalizeFloatTypes.cpp

```diff
@@ -2639,7 +2639,11 @@ SDValue DAGTypeLegalizer::PromoteFloatOp_FCOPYSIGN(SDNode *N, unsigned OpNo) {
 // Convert the promoted float value to the desired integer type
 SDValue DAGTypeLegalizer::PromoteFloatOp_UnaryOp(SDNode *N, unsigned OpNo) {
   SDValue Op = GetPromotedFloat(N->getOperand(0));
-  return DAG.getNode(N->getOpcode(), SDLoc(N), N->getValueType(0), Op);
+  if (N->getOpcode() == ISD::AssertNoFPClass)
+    return DAG.getNode(N->getOpcode(), SDLoc(N), N->getValueType(0), Op,
+                       N->getOperand(1));
+  else
+    return DAG.getNode(N->getOpcode(), SDLoc(N), N->getValueType(0), Op);
```

Comment:
Same here, doesn't really fi under unary op. Just make it its own PromoteFloatOp function